02-oop-inheritance/js/indexOOPClass.js

```diff
@@ -0,0 +1,256 @@
+        throw TypeError('Object prototype may only be an Object or null');
+      }
+
+      // 2. Let obj be the result of creating a new object as if by the
+      //    expression new Object() where Object is the standard built-in
+      //    constructor with that name
+      // 3. Set the [[Prototype]] internal property of obj to O.
+      Temp.prototype = O;
+      var obj = new Temp();
+      Temp.prototype = null; // Let's not keep a stray reference to O...
+
+      // 4. If the argument Properties is present and not undefined, add
+      //    own properties to obj as if by calling the standard built-in
+      //    function Object.defineProperties with arguments obj and
+      //    Properties.
+      if (arguments.length > 1) {
+        // Object.defineProperties does ToObject on its first argument.
+        var Properties = Object(arguments[1]);
+        for (var prop in Properties) {
+          if (hasOwn.call(Properties, prop)) {
+            obj[prop] = Properties[prop];
+          }
+        }
+      }
+
+      // 5. Return obj
+      return obj;
+    };
+  })();
+}
+
+
+// Function load ------------------------------------------------------------------------------------------------------------------------ /
+$(window).load(function() {	
+
+	var m1 = new MoviesModule.Movie(
+		"Lord of the Rings", 
+		"Film series consisting of three epic fantasy adventure films based on the novel The Lord of the Rings by J. R. R. Tolkien.",
+		"Peter Jackson"
+		);	
+		
+	var m2= new MoviesModule.Movie(
+		"Gladiator",
+		"Epic historical drama  general Maximus Decimus Meridius, who is betrayed when Commodus, the ambitious son of Emperor Marcus Aurelius, and search revenge",
+		 "Ridley Scott"
+		);
+
+	var m3 = new MoviesModule.Movie(
+		"A Beautiful Mind",
+		"A 2001 American biographical drama film based on the life of John Nash, a Nobel Laureate in Economics.",
+		"Ron Howard"
+		);
+
+	var m4 = new MoviesModule.DownloadableMovie(
+		"Batman Begun",
+			"The start of the new saga of Batman",
+			"Christopher Nolan"
+			);		
+
+	var peliculaElegida = prompt("Please select a movie\n1- Lord of the Rings (Play)\n2- Gladiator (Play)\n3- A Beautiful Mind (Play)\n4- Batman Begun (Download)");
+	var compartirCon = prompt("Share this film with ...");
+
+	switch(peliculaElegida){
+		case "1":
+			m1.play();
+			setTimeout(m1.stop(), 5000);
+			m1.share(m1, compartirCon);	
+			break;
+		case "2":
+			m2.play();
+			setTimeout(m2.stop(), 5000);
+			m2.share(m2, compartirCon);	
+			break;
+		case "3":
+			m3.play();
+			setTimeout(m3.stop(), 5000);
+			m3.share(m3, compartirCon);	
+			break;
+			case "4":
+			m4.download();
+			m4.play(); // Test to validate that Inheritance is working
+			m4.share(m4, compartirCon);
+			break;
+		default:
+			console.log("Wrong option");
+			break;
+	} // end switch	
+
+	// Practice - Point 12 - Adding actors to movie objects		
+	var actors = [];
+	var actor1 = new Actor("Vigo Mortensen", 20/10/1958);		
```

Comment:
I'm just curious, but are you sure you can pass a date like this? Don't you need to put it into a string or Date object?